storage/rocksdb/ha_rocksdb.cc

```diff
@@ -7841,6 +8001,20 @@ int ha_rocksdb::check_uniqueness_and_lock(
       return rc;
     }
 
+    /*
+      If the pk key has ttl, we may need to pretend the row wasn't
+      found if it is already expired. The pk record is read into
+      m_retrieved_record by check_and_lock_unique_pk().
+    */
+    if (is_pk(key_id, table, m_tbl_def) && found && m_pk_descr->has_ttl() &&
+        should_hide_ttl_rec(rocksdb::Slice(&m_retrieved_record.front(),
+                                           m_retrieved_record.size()),
+                            (row_info.tx->m_snapshot_timestamp
+                                 ? row_info.tx->m_snapshot_timestamp
```

Comment:
Yes, that is my logic.  All the other callsites of `should_hide_ttl_rec` are on the read path, and I believe that they should all have snapshots.  That's also why I have the logic to abort in `should_hide_ttl_rec` if curr_ts is 0.  I want to catch any cases where that may happen to verify that my logic is correct.   

This particular callsite is on the write path filtering side.  Here it's possible not to have a snapshot (for inserts).